plugins/quickinspector/quickoverlay.cpp

```diff
@@ -437,16 +439,16 @@ void QuickOverlay::windowAfterRendering()
         m_grabbedFrame.transform.translate(intersect.x() * m_renderInfo.dpr , -intersect.y() * m_renderInfo.dpr - h);
         m_grabbedFrame.image.setDevicePixelRatio(m_renderInfo.dpr);
 
-        if (!m_grabbedFrame.image.isNull()) {
-            m_sceneGrabbed.invoke(this, Qt::QueuedConnection, Q_ARG(GammaRay::GrabbedFrame, m_grabbedFrame));
-        }
+        Q_ASSERT(!m_grabbedFrame.image.isNull());
+        m_sceneGrabbed.invoke(this, Qt::QueuedConnection, Q_ARG(GammaRay::GrabbedFrame, m_grabbedFrame));
     }
 
     drawDecorations();
 
     m_window->resetOpenGLState();
 
     if (m_isGrabbingMode) {
+        locker.unlock();
         setIsGrabbingMode(false);
```

Comment:
Seeing this pattern also below, wouldn't it be better to assume the lock being held in setIsGrabbingMode? So the value doesn't change between the check here, and the call to it?